qiskit/algorithms/evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,245 @@
+            evolution_problem: Instance defining evolution problem. For the included Hamiltonian,
+                ``PauliOp``, ``SummedOp`` or ``PauliSumOp`` are supported by TrotterQRTE.
+
+        Returns:
+            Evolution result that includes an evolved state as a quantum circuit and, optionally,
+            auxiliary operators evaluated for a resulting state on a backend.
+
+        Raises:
+            ValueError: If ``t_param`` is not set to None in the EvolutionProblem (feature not
+                currently supported).
+            ValueError: If the ``initial_state`` is not provided in the EvolutionProblem.
+        """
+        evolution_problem.validate_params()
+        if evolution_problem.t_param is not None:
+            raise ValueError(
+                "TrotterQRTE does not accept a time dependent hamiltonian,"
+                "``t_param`` from the EvolutionProblem should be set to None."
+            )
+
+        if evolution_problem.aux_operators is not None and (
+            self._quantum_instance is None or self._expectation is None
+        ):
+            raise ValueError(
+                "aux_operators where provided for evaluations but no ``expectation`` or "
+                "``quantum_instance`` was provided."
+            )
+        hamiltonian = evolution_problem.hamiltonian
+        if not isinstance(hamiltonian, (PauliOp, PauliSumOp, SummedOp)):
+            raise ValueError(
+                f"TrotterQRTE only accepts PauliOp | "
+                f"PauliSumOp | SummedOp, {type(hamiltonian)} provided."
+            )
+        if isinstance(hamiltonian, OperatorBase):
+            hamiltonian = hamiltonian.bind_parameters(evolution_problem.hamiltonian_value_dict)
+        if isinstance(hamiltonian, SummedOp):
+            hamiltonian = self._summed_op_to_pauli_sum_op(hamiltonian)
+        # the evolution gate
+        evolution_gate = CircuitOp(
+            PauliEvolutionGate(hamiltonian, evolution_problem.time, synthesis=self._product_formula)
+        )
+
+        if evolution_problem.initial_state is not None:
+            initial_state = evolution_problem.initial_state
+            if isinstance(initial_state, QuantumCircuit):
+                initial_state = StateFn(initial_state)
+            evolved_state = evolution_gate @ initial_state
+
+        else:
+            raise ValueError("``initial_state`` must be provided in the EvolutionProblem.")
+
+        evaluated_aux_ops = None
+        if evolution_problem.aux_operators is not None:
+            evaluated_aux_ops = eval_observables(
+                self._quantum_instance,
+                evolved_state.primitive,
+                evolution_problem.aux_operators,
+                self._expectation,
+                evolution_problem.truncation_threshold,
+            )
+
+        return EvolutionResult(evolved_state, evaluated_aux_ops)
+
+    @staticmethod
+    def _summed_op_to_pauli_sum_op(
+        hamiltonian: Union[SummedOp],
```

Comment:
If this is always a SummedOp we don't need the `Union` 🙂 
```suggestion
        hamiltonian: SummedOp,
```